driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -196,6 +196,9 @@ public void operationComplete(ChannelFuture future) throws Exception {
         Futures.addCallback(initFuture, new FutureCallback<Void>() {
             @Override
             public void onSuccess(Void result) {
+                Host host = factory.manager.metadata.getHost(address);
+                if(host != null)
```

Comment:
Added a null check here and also in ef78e63 as I saw this raised in a test that decommissioned hosts.  Not 100% sure if this is the most appropriate fix.  I'm guessing this happens for connections belonging to a ControlConnection that had a removed Host?